(DNM) Trying out conda-smithy#2075

conda-forge-admin opened the pull request:
Hi! This is the friendly automated conda-forge-webservice.

I've started rerendering the recipe as instructed in #20.

If I find any needed changes to the recipe, I'll push them to this PR shortly. Thank you for waiting!



Here's a checklist to do before merging.
- [ ] Bump the build number [if needed](https://conda-forge.org/docs/maintainer/updating_pkgs.html#updating-recipes)

Fixes #20

conda-forge-webservices[bot] committed:
dummy commit for rerendering

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.
I do have some suggestions for making it better though...


For **recipe/meta.yaml**:

 * It looks like the '_libdav1d_api' output doesn't have any tests.

Commit:
MNT: Re-rendered with conda-build 24.9.0, conda-smithy 3.41.1, and conda-forge-pinning 2024.10.03.06.07.11

jaimergp committed:
MNT: Re-rendered with conda-build 24.9.0, conda-smithy 3.41.2.dev8+gca99aa5b, and conda-forge-pinning 2024.10.03.06.07.11"

jaimergp committed:
Enable micromamba

MNT: Re-rendered with conda-build 24.9.0, conda-smithy 3.41.2.dev8+gca99aa5b, and conda-forge-pinning 2024.10.03.06.07.11

jaimergp commented:
Timings for Miniforge

macOS, 1m8s:

```
2024-10-03T14:26:23.9738560Z + curl -L -O https://github.com/conda-forge/miniforge/releases/latest/download/Miniforge3-MacOSX-x86_64.sh
2024-10-03T14:27:31.8345340Z + echo -e '\n\nSetting up the condarc and mangling the compiler.'
```

Windows (in-script), 1m23s:

```
2024-10-03T14:26:29.9405199Z Downloading Miniforge
2024-10-03T14:26:32.9818273Z Installing Miniforge
2024-10-03T14:27:52.0805574Z Setting up configuration
```

Note I changed to D:\ for this PR and that already involves a significant cut!

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found some lint.

Here's what I've got...


For **recipe/meta.yaml**:

 * In conda-forge.yml: [`$.conda_install_tool`](https://conda-forge.org/docs/maintainer/conda_forge_yml/#conda-install-tool) `=` `pixi`.
    > 'pixi' is not valid under any of the given schemas
    <details>
    <summary>Schema</summary>

    ```json
    {
      "anyOf": [
        {
          "enum": [
            "conda",
            "mamba"
          ],
          "type": "string"
        },
        {
          "type": "null"
        }
      ],
      "default": "mamba",
      "title": "Conda Install Tool"
    }
    ```

    </details>

For **recipe/meta.yaml**:

 * It looks like the '_libdav1d_api' output doesn't have any tests.

jaimergp committed:
MNT: Re-rendered with conda-build 24.9.0, conda-smithy 3.41.2.dev8+gca99aa5b, and conda-forge-pinning 2024.10.03.06.07.11

jaimergp commented:
With micromamba, macOS takes 59s:

```
2024-10-03T15:04:20.9476240Z Downloading micromamba 1.5.10-0
2024-10-03T15:04:21.6084090Z Creating environment
2024-10-03T15:05:19.6315190Z + echo -e '\n\nSetting up the condarc and mangling the compiler.'
```

Windows, 2m41s (!!!), this is higher than expected.

```
2024-10-03T15:04:29.8468143Z Downloading micromamba 1.5.10-0
2024-10-03T15:04:31.0699440Z Creating environment
2024-10-03T15:07:12.5764284Z Setting up configuration
```

---

Another run:

macOS, 58s:

```
2024-10-03T15:25:31.1459260Z Downloading micromamba 1.5.10-0
2024-10-03T15:25:32.2555140Z Creating environment
2024-10-03T15:26:29.6806830Z Setting up the condarc and mangling the compiler.
```


Windows, 2m04s

```
2024-10-03T15:25:36.8743912Z Downloading micromamba 1.5.10-0
2024-10-03T15:25:38.0304293Z Creating environment
2024-10-03T15:27:40.5856277Z Setting up configuration
```

jaimergp committed:
error early

jaimergp committed:
MNT: Re-rendered with conda-build 24.9.0, conda-smithy 3.41.2.dev8+gca99aa5b, and conda-forge-pinning 2024.10.03.06.07.11

jaimergp commented:
Hm, the slowdowns on Windows could have been caused by us having the micromamba root on C:. Let's see what happens if we put it in D:.

jaimergp commented:
With D-only setups, micromamba can provision everything in 45s:

```
2024-10-03T15:48:54.1178298Z Downloading micromamba 1.5.10-0
2024-10-03T15:48:55.5904691Z Creating environment
2024-10-03T15:49:39.2868659Z ##[group]Configuring conda
```

jaimergp committed:
MNT: Re-rendered with conda-build 24.9.0, conda-smithy 3.41.2.dev8+gca99aa5b, and conda-forge-pinning 2024.10.15.19.09.52

jaimergp committed:
MNT: Re-rendered with conda-build 24.9.0, conda-smithy 3.41.2.dev8+gca99aa5b, and conda-forge-pinning 2024.10.15.19.09.52

jaimergp committed:
debug pkgs and downloads

jaimergp committed:
debug

jaimergp commented:
I can confirm the pkgs/ relocation works. See logs for details (DEBUG lines during the conda-build solves show that only some packages need to be fetched):

- [logs-downloads-are-cached-osx.txt](https://github.com/user-attachments/files/17386775/logs-downloads-are-cached-osx.txt)
- [logs-downloads-are-cached-windows.txt](https://github.com/user-attachments/files/17386776/logs-downloads-are-cached-windows.txt)

jaimergp committed:
MNT: Re-rendered with conda-build 24.9.0, conda-smithy 3.41.2.dev8+gca99aa5b, and conda-forge-pinning 2024.10.15.19.09.52

jaimergp commented:
Latest timings with micromamba and `D` drive:

OSX, 1min7s:

```
2024-10-15T23:37:41.5273930Z ##[group]Provisioning base env with micromamba
2024-10-15T23:38:54.5454540Z ##[group]Configuring conda
```

Windows, 53s:

```
2024-10-15T23:37:45.9497759Z ##[group]Provisioning base env with micromamba
2024-10-15T23:38:38.6859249Z ##[group]Configuring conda
```

jaimergp committed:
MNT: Re-rendered with conda-build 24.9.0, conda-smithy 3.41.2.dev8+gca99aa5b, and conda-forge-pinning 2024.10.17.13.07.12"

jaimergp committed:
write python_version for now

jaimergp committed:
hardcode for now

jaimergp committed:
MNT: Re-rendered with conda-build 24.9.0, conda-smithy 3.41.2.dev8+gca99aa5b, and conda-forge-pinning 2024.10.17.13.07.12

jaimergp committed:
debug to check where the packages are coming from

jaimergp committed:
MNT: Re-rendered with conda-build 24.9.0, conda-smithy 3.41.2.dev8+gca99aa5b, and conda-forge-pinning 2024.10.17.15.21.02

jaimergp committed:
try pixi

MNT: Re-rendered with conda-build 24.9.0, conda-smithy 3.41.2.dev8+gca99aa5b, and conda-forge-pinning 2024.10.17.15.21.02"

jaimergp committed:
MNT: Re-rendered with conda-build 24.9.0, conda-smithy 3.41.2.dev8+gca99aa5b, and conda-forge-pinning 2024.10.17.15.21.02

jaimergp committed:
MNT: Re-rendered with conda-build 24.9.0, conda-smithy 3.41.2.dev8+gca99aa5b, and conda-forge-pinning 2024.10.17.15.21.02

jaimergp committed:
MNT: Re-rendered with conda-build 24.9.0, conda-smithy 3.41.2.dev8+gca99aa5b, and conda-forge-pinning 2024.10.17.15.21.02

jaimergp committed:
MNT: Re-rendered with conda-build 24.9.0, conda-smithy 3.41.2.dev8+gca99aa5b, and conda-forge-pinning 2024.10.17.15.21.02

jaimergp committed:
retrigger

jaimergp committed:
MNT: Re-rendered with conda-build 24.9.0, conda-smithy 3.41.2.dev8+gca99aa5b, and conda-forge-pinning 2024.10.17.15.21.02

jaimergp committed:
MNT: Re-rendered with conda-build 24.9.0, conda-smithy 3.41.2.dev8+gca99aa5b, and conda-forge-pinning 2024.10.18.07.32.13

jaimergp committed:
MNT: Re-rendered with conda-build 24.9.0, conda-smithy 3.41.2.dev8+gca99aa5b, and conda-forge-pinning 2024.10.18.07.32.13

jaimergp committed:
MNT: Re-rendered with conda-build 24.9.0, conda-smithy 3.41.2.dev8+gca99aa5b, and conda-forge-pinning 2024.10.18.07.32.13

jaimergp committed:
MNT: Re-rendered with conda-build 24.9.0, conda-smithy 3.41.2.dev8+gca99aa5b, and conda-forge-pinning 2024.10.18.09.40.47

jaimergp committed:
MNT: Re-rendered with conda-build 24.9.0, conda-smithy 3.41.2.dev8+gca99aa5b, and conda-forge-pinning 2024.10.18.09.40.47

jaimergp committed:
MNT: Re-rendered with conda-build 24.9.0, conda-smithy 3.41.2.dev8+gca99aa5b, and conda-forge-pinning 2024.10.18.09.40.47

jaimergp committed:
MNT: Re-rendered with conda-build 24.9.0, conda-smithy 3.41.2.dev8+gca99aa5b, and conda-forge-pinning 2024.10.18.09.40.47

jaimergp committed:
MNT: Re-rendered with conda-build 24.9.0, conda-smithy 3.41.2.dev8+gca99aa5b, and conda-forge-pinning 2024.10.18.09.40.47

jaimergp committed:
MNT: Re-rendered with conda-build 24.9.0, conda-smithy 3.41.2.dev8+gca99aa5b, and conda-forge-pinning 2024.10.18.09.40.47

jaimergp commented:
Timings for Pixi:

- Linux: 10s 😲 (we reuse the Docker container cached Miniforge)

```
2024-10-18T11:02:46.9653537Z + curl -fsSL https://pixi.sh/install.sh
2024-10-18T11:02:56.4295577Z + setup_conda_rc /home/conda/feedstock_root /home/conda/recipe_root /home/conda/feedstock_root/.ci_support/linux_64_.yaml
```

- macOS: 53s

```
2024-10-18T11:02:18.6818740Z ##[group]Provisioning base env with pixi
2024-10-18T11:03:11.8383210Z ##[group]Configuring conda
```

- Windows: 60s

```
2024-10-18T11:02:19.4140607Z ##[group]Provisioning base env with pixi
2024-10-18T11:03:19.5561424Z ##[group]Configuring conda

```
